### PR TITLE
fix: Apply external refresh segments at job level

### PR DESCRIPTION
**Pull request by weiliu1031**
issue: #49478
issue: #45881

### What changed
- Persist each external refresh task's kept and updated segment result in task meta.
- Merge all finished task results once at job level before persisting the refresh job as Finished.
- Keep progress from exposing Finished until segment apply and job Finished persistence are complete.
- Clear task result payloads after a successful job commit.
- Reject duplicate updated segment IDs during job-level merge.

### Why
A multi-task external table refresh previously applied each task response as if it were the collection-global segment set. Later task completions could drop sibling task segments, so the API returned RefreshCompleted with only partial row_count.

### Verification
- git diff --check

### CODE REVIEW

**Comment by liliu-z** [comment]
**internal/datacoord/external_collection_refresh_inspector.go:0** New version persists segment results (kept_segments/updated_segments) into task proto via `UpdateResultWithMeta`, then relies on `applyJobInfo` in the checker to apply them at job level. If the cluster is rolled back to an older version after some tasks reach Finished state:

1. Old inspector skips Finished tasks (inspector.go:113-116) and won't re-enqueue them.
2. Old checker has no `applyJobInfo` callback — it only calls `UpdateJobState`, not `UpdateJobStateWithPreApply`.
3. Old version does not recognize `kept_segments`/`updated_segments` fields.

Result: segment results embedded in Finished tasks become permanently orphaned. This should be explicitly documented as forward-only in release notes.

**Comment by liliu-z** [comment]
**internal/metastore/kv/datacoord/kv_catalog.go:1150** `SaveExternalCollectionRefreshTask` (line 1150) does a single-key `proto.Marshal(task)` write to etcd with no size guard, splitting, or compression. Each task now embeds `SegmentInfo` protos with fake binlogs (`buildFakeBinlogs` at task_update.go:736 produces one FieldBinlog with one Binlog entry per segment). Per-segment overhead is ~200-500 bytes, so 100 segments/task ≈ 50KB — well under etcd's 1.5MB limit in normal scenarios. However, schemas with hundreds of fields could push `ChildFields` lists higher. Risk is low but this is a new capacity growth surface with no upper-bound enforcement.

File: internal/datacoord/external_collection_refresh_meta_test.go [hunk 1]
```diff
@@ -664,6 +664,96 @@ func TestExternalCollectionRefreshMeta_UpdateTaskState(t *testing.T) {
 	})
 }
 
+func TestExternalCollectionRefreshMeta_UpdateTaskResult(t *testing.T) {
```
Comment (liliu-z) [inline internal/datacoord/external_collection_refresh_meta_test.go:739]: These three methods are only covered indirectly through checker integration tests (`TestAggregateJobState_FinishedApplyOnce`, `TestAggregateJobState_ClearsTaskResultsAfterFinishedPersisted`). Missing dedicated unit tests for:
- `preApply` callback failure causing job to transition to Failed
- `preApply` success but `SaveJob` failure (retry/idempotency path)
- `ClearTaskResultsByJobID` partial failure when some tasks fail to save

No correctness bug, but reduced confidence in edge-case behavior.

**Comment by liliu-z** [comment]
**internal/datacoord/checker_test.go:334** Test uses `time.Sleep` for synchronization instead of a deterministic mechanism (e.g., channel, condition variable, or polling with timeout). Under CI load this can lead to intermittent failures.
